Form/Type/MoneyType.php

```diff
@@ -31,18 +31,22 @@ public function __construct(
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->add('tbbc_amount', new TextType())
+            ->add('tbbc_amount', 'Symfony\Component\Form\Extension\Core\Type\TextType')
             ->add('tbbc_currency', $options['currency_type'])
             ->addModelTransformer(
                 new MoneyToArrayTransformer($this->decimals)
             );
     }
 
+    /**
+     * @param OptionsResolver $resolver
+     */
     public function configureOptions(OptionsResolver $resolver)
     {
         $resolver
             ->setDefaults(array(
-                'currency_type' => 'tbbc_currency',
+                'data_class' => null,
```

Comment:
@Baptouuuu The way this form works is by you setting `Money` object as a value and it will convert it to array of price and value internally for rendering 2 fields. If `data_class` is not provided it would still try to treat converted array as Money class and will fail.